tests/perturbator/test_perturbator.py

```diff
@@ -201,6 +203,93 @@ def test_stochastic_relative_perturbate_scale_provided_is_used(average_effect, s
     assert np.var(perturbated_values) == np.var(effect)
 
 
+@pytest.mark.parametrize("average_effect", [0.1, 0.04])
+def test_beta_relative_perturbate(average_effect):
+    # given
+    range_min = -0.8
+    range_max = 5
+    pert = BetaRelativePerturbator(range_min=range_min, range_max=range_max)
+
+    average_effect_inv_transf = (average_effect - range_min) / (range_max - range_min)
+    scale_inv_transf = average_effect_inv_transf
+    a = average_effect_inv_transf / (scale_inv_transf * scale_inv_transf)
+    b = (1 - average_effect_inv_transf) / (scale_inv_transf * scale_inv_transf)
+
+    np.random.seed(24)
+    beta = np.random.beta(a / abs(average_effect), b / abs(average_effect), 2)
+    beta_transf = beta * (range_max - range_min) + range_min
+
+    effect = (1 + beta_transf) * continuous_df.query("treatment == 'B'")[
+        "target"
+    ].values
+
+    # when
+    np.random.seed(24)
+    perturbated_values = (
+        pert.perturbate(continuous_df, average_effect=average_effect)
+        .query("treatment == 'B'")["target"]
+        .values
+    )
+
+    # then
+    assert np.mean(perturbated_values) == np.mean(effect)
+    assert np.var(perturbated_values) == np.var(effect)
+
+
+@pytest.mark.parametrize("average_effect", [0.1, 0.04])
+def test_clustered_beta_relative_perturbate(average_effect):
+    range_min = -0.8
+    range_max = 5
+
+    df_clustered = generate_clustered_data()
+    pert = ClusteredBetaRelativePerturbator(
+        range_min=range_min, range_max=range_max, cluster_cols=["city_code"]
+    )
+    df_pert = pert.perturbate(
+        pd.concat([df_clustered for _ in range(100)]), average_effect=average_effect
```

Comment:
do you have some kind of test that, when applying a certain average effect (let's say +20%), you have data in treatment with the target (on average) higher than the data in control by a 20%? if not, wdyt about adding it?

Reply:
Not sure I understand, cause I think that's exactly what this test does^^
The only catch is that I'm allowing for a given `average_effect`, I'm allowing the _observed_ average effect in the treatment to vary by +-50% from the given `average_effect`. I realize that this sounds like a lot, but given the small size of the test df and the groups in that df, the observed average effect can vary quite a bit (which is kind of the point).

Reply:
alright, good. it was not clear to me that this was what we were testing, but great then :)

Reply:
looks like a lot variability. Shouldn't it converge to 0 if repeated many times? Maybe you could do this, iterate and check that in average it's smaller than epsilon. WDYT?

Reply:
@pablobd Good point, I changed it now to iterate a few times to decrease the variability. Not doing anything more fancy, like checking that the variability decreases with more runs, also because it slows down the unit tests. But let me know if you think it's important and I can do it.